nexus/types/src/deployment/execution/dns.rs

```diff
@@ -164,16 +170,36 @@ pub fn blueprint_external_dns_config<'a>(
     external_dns_zone_name: String,
 ) -> DnsConfigZone {
     let nexus_external_ips = blueprint_nexus_external_ips(blueprint);
+    let dns_external_ips = blueprint_external_dns_resolver_ips(blueprint);
 
-    let dns_records: Vec<DnsRecord> = nexus_external_ips
+    let nexus_dns_records: Vec<DnsRecord> = nexus_external_ips
         .into_iter()
         .map(|addr| match addr {
             IpAddr::V4(addr) => DnsRecord::A(addr),
             IpAddr::V6(addr) => DnsRecord::Aaaa(addr),
         })
         .collect();
 
-    let records = silos
+    let mut zone_records: Vec<DnsRecord> = Vec::new();
+    let external_dns_records: Vec<(String, Vec<DnsRecord>)> = dns_external_ips
```

Comment:
Should we sort these for consistency?

Reply:
The only thing I'm trying to avoid is having the contents of NS records change because of some non-determinism here, causing either blueprint or DNS generation bumps under normal conditions even when there's been no changes.  I think that's important to avoid but any deterministic order seems fine.

Reply:
https://github.com/oxidecomputer/omicron/pull/8047/commits/a99660d4aaec18b5a089f03ede4510c8df168fda (the case i was thinking about, something unevenly loading the first nameserver it sees, wouldn't really be helped here anyway. the order is basically stable in practice, since if this was nondeterministic it'd only change when the configuration is otherwise made to change. that's not a load balancing mechanism in any case.)